src/store.ts

```diff
@@ -29,15 +49,57 @@ export function loadLocalJwtStore(): JwtStore {
 
   return {
     findJwtForRoom: (roomName) => confabs.JWTs[roomName],
+
     findRefreshTokenForRoom: (roomName) => confabs.refresh[roomName],
+
     storeJwtForRoom: (roomName, encodedJwt, encodedRefreshToken) => {
       confabs.JWTs[roomName] = encodedJwt;
       if (encodedRefreshToken) {
         confabs.refresh[roomName] = encodedRefreshToken;
       }
       saveToStorage(confabs);
     },
+
     isNewMonthlyActiveUser: () => performMauCheck(confabs),
+
+    availableRecordings: () => confabs.recordings,
+
+    findRecordingAtURL: (url: string) => confabs.recordings[url],
+
+    upsertRecordingForRoom: (
+      url: string,
+      roomName: string,
+      ttl: number | undefined
+    ) => {
+      console.log(
+        "!!! upsertRecording " +
+          url +
+          " for room " +
+          roomName +
+          " ttl=" +
+          ttl +
+          "  createP=" +
+          !confabs.recordings[url]
+      );
+
+      const now = Math.ceil(new Date().getTime() / 1000);
+      if (typeof ttl === "undefined") {
+        ttl = 24 * 60 * 60;
+      }
+      const expiresAt = now + ttl;
```

Comment:
does the server calculate expiration separately from this? otherwise a client who wants their recording to persist for 24+ hours can set `expiresAt` to whatever they want

Reply:
The 24h is hard-wired into the 8x8 back-end. i have asked them to add a field in the API response to indicate the TTL, so if/when it changes in the future, reality will be enforced...